pkg/declarativeconfig/context.go

```diff
@@ -0,0 +1,26 @@
+package declarativeconfig
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/generated/storage"
+)
+
+type originCheckerKey struct{}
+
+// AllowOnlyDeclarativeOperations signals that the context holder is allowed to modify declarative resources.
+const AllowOnlyDeclarativeOperations = true
+
+// WithAllowOnlyDeclarativeOperations returns a context that is a child of the given context and allows to modify
+// declarative resources.
+func WithAllowOnlyDeclarativeOperations(ctx context.Context) context.Context {
```

Comment:
From the naming, maybe `WithModifyDeclarativeResource` is better? I think we can drop the only, think it's more confusing then really helpful but up to you.